src/Nox.Types.EntityFramework/Abstractions/INoxDatabaseConfigurator.cs

```diff
@@ -1,9 +1,14 @@
-using Microsoft.EntityFrameworkCore.Metadata.Builders;
 using Nox.Solution;
+using Nox.Types.EntityFramework.EntityBuilderAdapter;
 
 namespace Nox.Types.EntityFramework.Abstractions;
 
 public interface INoxDatabaseConfigurator
 {
-    void ConfigureEntity(NoxSolutionCodeGeneratorState codeGeneratorState, EntityTypeBuilder builder, Entity entity, IReadOnlyList<EntityRelationshipWithType> relationshipsToCreate);
+    void ConfigureEntity(
+        NoxSolutionCodeGeneratorState codeGeneratorState,
+        IEntityBuilderAdapter builder,
+        Entity entity,
+        NoxSolution noxSolution,
+        Func<string, Type?> getTypeByNameFunc);
```

Comment:
lets discuss this Func<string, Type?> getTypeByNameFunc
is kind of obscure, we need to think about this

Reply:
the codeGeneratorState contains the function no need to input it here, correct?

Reply:
removed it completely